lib/ecto/query/builder.ex

```diff
@@ -82,7 +82,15 @@ defmodule Ecto.Query.Builder do
     {{:{}, [], [:fragment, [], merge_fragments(pieces, frags)]}, params_acc}
   end
 
-  def escape({:unsafe_fragment, _, [{:^, _, [var]} | frags]}, _type, params_acc, vars, env) do
+  def escape({:unsafe_fragment, _, [{:^, _, [var]}]}, _type, params_acc, _vars, _env) do
+    expr = quote do
+      Ecto.Query.Builder.unsafe_fragment!(unquote(var))
```

Comment:
We don't need to introduce a new function, just pass an empty list as second argument.